metric_descriptions/novelty.yml

```diff
@@ -1,46 +1,46 @@
-    Initially, the Novelty (EFD) is defined as: 
-    $$EFD=C\sum_{i_k \in \mathbb{R}}^{}disk(k)p(rel \mid i_k,u)(-\log_2 p(i \mid seen,\theta))$$where, \(disk(k)\) is a discount function, \(p(rel)\) is the probability of relevance, and the \(\log_{2}\) factor is a relative measure of how likely items are to be seen with respect to each other.
-    In RS Metrics the computation assumes that (i) recommendation items are equally weighted (rank=1), (ii) no extensive information is preserved if an item is liked, picked, enjoyed (relevance=1), (iii) user views all recommendation items together -without paging- (discount=1). See paper \(\href{https://dl.acm.org/doi/abs/10.1145/2043932.2043955}{https://dl.acm.org/doi/abs/10.1145/2043932.2043955}\). Thus, the computation is determined by the following formula: 
-    $$Novelty=\frac{\sum_{u}^{rec\_users}nov\_user(u)}{rec\_users}$$where, $$nov\_user(u)=\frac{\sum_{i}^{services(u)}novelty(i)}{recommendations(u)}$$and,
-    \[ novelty(i)= \begin{cases} -\log_2\left(\frac{count(i)}{recommendations}\right),& i \in services(u) \\ -\log_2\left(\frac{min\{count(j), \forall j \in services\}}{recommendations}\right), & i \notin services(u) \end{cases} \]
+    Calculating novelty of the recommender system based on the user actions using the following formula:
 
+    $$Novelty=\frac{\sum\nolimits_{i \in R}-log(p(i))}{|R|}$$
+
+    For each service item $$i$$ beloning to the set of recommended services $$R$$ calculate the portion $$p(i)$$ of the times
+    the service has beend viewd to the total views of the services produced by the user actions data.
 
 
 
 output:
     type: float
     min: 0
     max: +\(\infty\)
-    comment: Novelty of a user is 0 when a single item is always chosen or recommended, and \(n\log_2{n}\) when n items are chosen or recommended equally often.
+    comment: Novelty expresses the ability of the system to recommend items that are novel - not seen before by the population of users. A smaller number expresses that more services are being recommended that the users have not seen before
 
 prerequisites:
     - all available recommendations associated with registered users
-    - all available user actions associated with registered users
+    - a subset of the available user actions associated with registered users that expresses transitions to service pages
 
 process:
     - step: Clean up
       details: >
         Recommendations and user actions clean up; entries removal where users or services are not found in "users" or "services" files accordingly
-    - step: Services Impact
+    - step: User actions that target services
       details: >
-        Calculation of the impact of the services, by counting how many times each service (i.e. i) was suggested to all possible users: count(i)
-    - step: Calculate novelty of a service (i.e. novelty(i))
+        Identify and keep user actions that express transition to target pages that are views of services
```

Comment:
Additionally, user actions where the source and the target page belong to the same service's space are removed from the process.

Reply:
done